linkerd/metrics/src/gauge.rs

```diff
@@ -1,50 +1,44 @@
-use std::fmt::{self, Display};
-
-use tracing::warn;
-
 use super::{FmtLabels, FmtMetric};
+use std::fmt::{self, Display};
+use std::sync::atomic::{AtomicU64, Ordering};
 
 /// An instaneous metric value.
-#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
-pub struct Gauge(u64);
+#[derive(Debug, Default)]
+pub struct Gauge(AtomicU64);
 
 impl Gauge {
     /// Increment the gauge by one.
-    pub fn incr(&mut self) {
-        if let Some(new_value) = self.0.checked_add(1) {
-            (*self).0 = new_value;
-        } else {
-            warn!("Gauge overflow");
-        }
+    pub fn incr(&self) {
+        self.0.fetch_add(1, Ordering::SeqCst);
     }
 
     /// Decrement the gauge by one.
-    pub fn decr(&mut self) {
-        if let Some(new_value) = self.0.checked_sub(1) {
-            (*self).0 = new_value;
-        } else {
-            warn!("Gauge underflow");
-        }
+    pub fn decr(&self) {
+        self.0.fetch_sub(1, Ordering::SeqCst);
+    }
```

Comment:
I don't think these need to be `SeqCst`. This means that they will be part of a global order with _other_ atomic values, which I don't think is really necessary. They probably only need to be `Release`, so that any loads in calls to `Gauge::value` will happen after the increment/decrement ops.